src/tmx/TmxUtils/test/J2735MessageTest.cpp

```diff
@@ -542,6 +542,106 @@ TEST_F(J2735MessageTest, EncodeBasicSafetyMessage)
+	auto partIICnt = (PartIIcontent_t*) calloc(1, sizeof(PartIIcontent_t));
+	partIICnt->partII_Id = 2;
+	partIICnt->partII_Value.present = partII_Value_PR_SpecialVehicleExtensions;
+	auto specialVEx= (SpecialVehicleExtensions_t*) calloc(1, sizeof(SpecialVehicleExtensions_t));
+	auto emergencyDetails= (EmergencyDetails_t*) calloc(1, sizeof(EmergencyDetails_t));
+	emergencyDetails->lightsUse = LightbarInUse_inUse;
+	auto resp_type = (ResponseType_t*) calloc(1, sizeof(ResponseType_t));
+	*resp_type = ResponseType_emergency;
+	emergencyDetails->responseType = resp_type;
+	emergencyDetails->sirenUse = SirenInUse_inUse;	
+	specialVEx->vehicleAlerts = emergencyDetails;
+	partIICnt->partII_Value.choice.SpecialVehicleExtensions = *specialVEx;
+    asn_sequence_add(&bsmPartII->list.array, partIICnt);
+	message->partII = bsmPartII;
+
+	// BSM regional extension
+    auto regional = (BasicSafetyMessage::BasicSafetyMessage__regional *)calloc(1, sizeof(BasicSafetyMessage::BasicSafetyMessage__regional));
+    auto reg_bsm = (Reg_BasicSafetyMessage *)calloc(1, sizeof(Reg_BasicSafetyMessage));
+    reg_bsm->regionId = 128;
+    reg_bsm->regExtValue.present = Reg_BasicSafetyMessage__regExtValue_PR_BasicSafetyMessage_addGrpCarma;
+
+    auto carma_bsm_data = (BasicSafetyMessage_addGrpCarma_t *)calloc(1, sizeof(BasicSafetyMessage_addGrpCarma_t));
+    auto carma_bsm_destination_points = (BasicSafetyMessage_addGrpCarma::BasicSafetyMessage_addGrpCarma__routeDestinationPoints *)calloc(1, sizeof(BasicSafetyMessage_addGrpCarma::BasicSafetyMessage_addGrpCarma__routeDestinationPoints));
+    auto point = (Position3D_t *)calloc(1, sizeof(Position3D_t));
+    point->lat = 12;
+    point->Long = 1312;
+    asn_sequence_add(&carma_bsm_destination_points->list.array, point);
+    auto point2 = (Position3D_t *)calloc(1, sizeof(Position3D_t));
+    point2->lat = 1222;
+    point2->Long = 1312323;
+    asn_sequence_add(&carma_bsm_destination_points->list.array, point2);
+    carma_bsm_data->routeDestinationPoints = carma_bsm_destination_points;
+    reg_bsm->regExtValue.choice.BasicSafetyMessage_addGrpCarma = *carma_bsm_data;
+
+    asn_sequence_add(&regional->list.array, reg_bsm);
+    message->regional = regional;
+	tmx::messages::BsmEncodedMessage bsmEncodeMessage;
+	tmx::messages::BsmMessage*  _bsmMessage = new tmx::messages::BsmMessage(message);
+	tmx::messages::MessageFrameMessage frame_msg(_bsmMessage->get_j2735_data());
+	bsmEncodeMessage.set_data(TmxJ2735EncodedMessage<BasicSafetyMessage>::encode_j2735_message<codec::uper<MessageFrameMessage>>(frame_msg));
+	free(message);
+	free(frame_msg.get_j2735_data().get());
+	ASSERT_EQ(20,  bsmEncodeMessage.get_msgId());
+	std::string expectedBSMEncHex = "00143e604043030280ffdbfba868b3584ec40824646400320032000c888fc834e37fff0aaa960fa0080d0824088012486b49d218d693ae3e1ad276e335aeec2100";
+	ASSERT_EQ(expectedBSMEncHex, bsmEncodeMessage.get_payload_str());
```

Comment:
In addition to checking that the BSM encoded hex matches the expected value, would it also make sense to check some of the other fields in the encoded message (similar to the `ASSERT_EQ(20,  bsmEncodeMessage.get_msgId());` line) in order to ensure those values were encoded properly as well? Perhaps just the Part II and regional extension fields that were set?